packages/google-pubsub-microservice/README.md

```diff
@@ -0,0 +1,88 @@
+<p align="center">
+  <a href="http://nestjs.com"><img src="https://nestjs.com/img/logo_text.svg" alt="Nest Logo" width="320" /></a>
+</p>
+
+<p align="center">
+  A <a href="https://github.com/nestjs/nest">Nest</a> Google PubSub Microservice.
+</p>
+
+# NestJS Google Cloud PubSub Micro-service
+
+A custom [NestJS Microservice](https://docs.nestjs.com/microservices/basics) transport strategy using [Google Cloud PubSub](https://cloud.google.com/pubsub/docs).
+
+## Installation
+
+```bash
+npm install --save @algoan/nestjs-google-pubsub-microservice
+```
+
+## Usage
+
+To start a new Google Cloud PubSub server:
+
+**Server setup**:
+
+```typescript
+// main.ts
+import { INestMicroservice } from '@nestjs/common';
+import { NestFactory } from '@nestjs/core';
+import { GCPubSubServer } from '@algoan/nestjs-google-pubsub-microservice':
+import { AppModule } from './app.module';
+
+async function bootstrap() {
+  const options: GooglePubSubOptions = {
+    projectId: 'test',
+    subscriptionsPrefix: 'test-app',
+  }
+
+  const app: INestMicroservice = await NestFactory.create(AppModule, {
+    strategy
+  })
+
+  await app.listen(() => console.log('Server running!'));
+}
+bootstrap()
+```
+
+**Controller**:
+
+```typescript
+import { EmittedMessage } from '@algoan/pubsub';
+import { Controller } from '@nestjs/common';
+import { EventPattern, Payload } from '@nestjs/microservices';
+import { AppService } from './app.service';
+
+@Controller()
+export class AppController {
+  constructor(private readonly appService: AppService) {}
+
+  /**
+   * Handle the test event
+   * @param data Payload sent
+   */
+  @EventPattern('test_event')
+  public async handleTestEvent(@Payload() data: EmittedMessage<{ hello: string }>): Promise<void> {
+    /**
+     * Handle data emitted by Google PubSub
+     */
+    this.appService.handleTestEvent(data);
+  }
+}
+```
+
+## API
+
+This module uses [@algoan/pubsub](https://github.com/algoan/pubsub) library, by default, automatically acknowledges emitted messages.
+
+### `GCPubSubServer(options)`
+
+Create a new Server instance of Google PubSub. It retrieves all message handlers patterns and creates [subscriptions](https://cloud.google.com/pubsub/docs/pull).
+
+- `options`: Algoan PubSub options. More information [here](https://github.com/algoan/pubsub/#pubsubfactorycreate-transport-options-).
+
+## Other node modules
```

Comment:
`Alternate node modules` might be better? I am not sure 🤔

Reply:
Yes I should find a better header... Maybe `Other NestJS Google PubSub Server` ?

Reply:
Yes that's better